tungsten_ci_utils/nexus_cleanup/clean_up.groovy

```diff
@@ -0,0 +1,97 @@
+import org.sonatype.nexus.repository.storage.StorageFacet;
+import org.sonatype.nexus.common.app.GlobalComponentLookupHelper;
+import org.sonatype.nexus.repository.maintenance.MaintenanceService;
+import org.sonatype.nexus.repository.storage.ComponentMaintenance;
+import org.sonatype.nexus.repository.storage.Query;
+import org.sonatype.nexus.script.plugin.RepositoryApi;
+import org.sonatype.nexus.script.plugin.internal.provisioning.RepositoryApiImpl;
+import com.google.common.collect.ImmutableList;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+def retentionDays = 15;
+def retentionBorder = 15;
+def repositoryName = 'BartsDockerRepo';
+def whitelisted_tag_suffixes = ["ocata", "newton", "latest", "40", "94", "122", "129", "161", "214", "309", "360"].toArray();
+def retentionCounter = 0;
+log.info(":::Cleanup script started!");
+MaintenanceService service = container.lookup("org.sonatype.nexus.repository.maintenance.MaintenanceService");
+def repo = repository.repositoryManager.get(repositoryName);
+def tx = repo.facet(StorageFacet.class).txSupplier().get();
+def components = null;
+try {
+    tx.begin();
+    components = tx.browseComponents(tx.findBucket(repo));
+} catch (Exception e) {
+    log.info("Error: " + e);
+} finally {
+    if (tx != null)
+        tx.close();
+}
+
+if (components != null) {
+    def retentionDate = DateTime.now().minusDays(retentionDays).dayOfMonth().roundFloorCopy();
+    int deletedComponentCount = 0;
+    int compCount = 0;
+    def listOfComponents = ImmutableList.copyOf(components);
+    def previousComp = listOfComponents.head().name();
+    def checkValue = null;
+    retentionCounter = 0;
+    def prev = null;
+    listOfComponents.reverseEach {  comp ->
+
+        def splited = comp.version();
+        def spl = splited.split("-");
+
+        for (i = spl.length - 1; i < spl.length; i++) {
+            checkValue = null;
+            for (j = 0; j < whitelisted_tag_suffixes.length; j++) {
+                Pattern pattern = Pattern.compile(whitelisted_tag_suffixes[j]);
+                Matcher matcher = pattern.matcher(spl[i]);
+                boolean found = matcher.matches();
+                if (found == true) {
```

Comment:
Correct me if I'm mistaken but looking at the current whitelist this condition may be just an exact match if (spl[i] == whitelisted_tag_suffixes[j])